app/models/order_import.rb

```diff
@@ -131,8 +131,15 @@ def handle_save_nothing_on_error # TODO refactor and rename
   end
 
   def import_row(row) # TODO refactor
-    row_importer = OrderRowImporter.new(row, self)
-    row_importer.import
+    begin
+      row_importer = OrderRowImporter.new(row, self)
+      row_importer.import
+    rescue => e
+      ActiveSupport::Notifications.instrument('external_error',
+        exception: e, information: "Failed to bulk import: #{upload_file_path}")
+      row_importer.add_error("Failed to import row")
+      set_error_mode
```

Comment:
Is this `set_error_mode` necessary since the line above it adds the error, and the next conditional checks for errors?